src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -216,20 +216,34 @@ public override void Complete(Exception exception = null)
 
             _internalTokenSource?.Dispose();
 
-            BufferSegment segment = _head;
-            while (segment != null)
+            FlushInternal();
```

Comment:
Is there any point to call Flush if _leaveOpen is false? Same with FlushAsync.

Reply:
I guess if Dispose noops, then we still care about flushing.

Reply:
Flush returns rented memory as well as flushed if that memory is non empty